openlibrary/templates/type/user/view.html

```diff
@@ -41,7 +41,7 @@ <h1>$page.displayname</h1>
 
     <h2>$_('Reading Log')</h2>
     $if settings and settings.get('public_readlog', 'no') == "yes":
-      <p>$:_('This user has chosen to publicly share the books they are <a href="$(page.key)/books/currently-reading">currently reading</a>, <a href="$(page.key)/books/already-read">have already read</a>, and <a href="$(page.key)/books/want-to-read">want to read</a>!')</p>
+      <p>$_('This user has chosen to publicly share the books they are ')<a href="$(page.key)/books/currently-reading">$_('currently reading')</a>, <a href="$(page.key)/books/already-read">$_('have already read')</a>, $_('and') <a href="$(page.key)/books/want-to-read">$_('want to read')</a>$_('!')</p>
```

Comment:
@cdrini. @tfmorris  I though the `:` was supposed to allow this to work, although I can't quickly find documentation on the mechanism.

Would a better way to do this be:
`$_('This user has chosen to publicly share the books they are <a href="{username}/books/currently-reading">currently reading</a>, ... blah <a href="{username}">blah</a> blah').format(username=page.key)`  and then we could use this style for all strings which insert untranslatable variables.

Reply:
@hornc thought the same thing too would love feedback on this from @cdrini and @tfmorris

Reply:
The `:` allows the markup through, but it doesn't fix the nested substitution of $(page.key), which is what is not working here. There is a built-in substitution mechanism in gettext which could be used with parameters like %(username)s, but a better solution would be to pull out the markup altogether like @cdrini did in one of his recent fixes.

Reply:
Yep; as others mentioned, splitting the translation string up like this is an i18n anti-pattern, because translating parts of a sentence and concatenating them is very rarely going to result in an understandable translation (mathematically, the "translation function" is not [additive](https://en.wikipedia.org/wiki/Additive_map) :P) I'm not sure what the best pattern going forward is (I have at least 2 possible patterns in mind), but for this specific case, we should just use relative URLs in the anchor hrefs; we don't need the vars at all:

```html
<p>$:_('This user has chosen to publicly share the books they are <a href="books/currently-reading">currently reading</a>, <a href="books/already-read">have already read</a>, and <a href="books/want-to-read">want to read</a>!')</p>
```

Then do a "Find All" in VS Code for a substring (e.g. "user has chosen to publicly share"), and update the the i18n files to use the updated source string.

Reply:
^ Context: My suggestion won't work, because it's relative to the _parent_ dir, not current file :P